src/generators.rs

```diff
@@ -985,6 +985,109 @@ pub fn directed_grid_graph(
     })
 }
 
+/// Generate an undirected hexagonal graph.
+///
+/// :param int rows: The number of rows to generate the graph with.
+/// :param list cols: The number of rows to generate the graph with.
+/// :param bool multigraph: When set to False the output
+///     :class:`~retworkx.PyGraph` object will not be not be a multigraph and
+///     won't  allow parallel edges to be added. Instead
+///     calls which would create a parallel edge will update the existing edge.
+///
+/// :returns: The generated hexagonal graph
+/// :rtype: PyGraph
+/// :raises IndexError: If neither ``rows`` or ``cols`` are
+///      specified
+///
+/// .. jupyter-execute::
+///
+///   import os
+///   import tempfile
+///
+///   import pydot
+///   from PIL import Image
+///
+///   import retworkx.generators
+///
+///   graph = retworkx.generators.hexagonal_graph(2, 2)
+///   dot_str = graph.to_dot(
+///       lambda node: dict(
+///           color='black', fillcolor='lightblue', style='filled'))
+///   dot = pydot.graph_from_dot_data(dot_str)[0]
+///
+///   with tempfile.TemporaryDirectory() as tmpdirname:
+///       tmp_path = os.path.join(tmpdirname, 'dag.png')
+///       dot.write_png(tmp_path)
+///       image = Image.open(tmp_path)
+///       os.remove(tmp_path)
+///   image
+///
+#[pyfunction(multigraph = true)]
+#[text_signature = "(/, rows=None, cols=None, multigraph=True)"]
+pub fn hexagonal_graph(
+    py: Python,
+    rows: Option<usize>,
+    cols: Option<usize>,
+    multigraph: bool,
+) -> PyResult<graph::PyGraph> {
+    let mut graph = StableUnGraph::<PyObject, PyObject>::default();
+
+    if rows.is_none() || cols.is_none() {
+        return Err(PyIndexError::new_err(
+            "dimensions and weights list not specified",
+        ));
+    }
+
+    let mut rowlen = rows.unwrap_or(0);
+    let mut collen = cols.unwrap_or(0);
+
+    // Needs two times the number of nodes vertically
+    rowlen = 2 * rowlen + 2;
+    collen += 1;
+    let num_nodes = rowlen * collen;
+
+    let nodes: Vec<NodeIndex> =
+        (0..num_nodes).map(|_| graph.add_node(py.None())).collect();
+
+    // Add column edges
+    for i in 0..collen {
+        for j in 0..(rowlen - 1) {
+            if i + 1 < rowlen {
```

Comment:
I don't believe this check `if i + 1 < rowlen` is needed, e.g `hexagonal_graph(1, 3)` will give wrong result.

Reply:
Oh, thanks. I forgot that I added `rowlen - 1` on the for loop.